src/openvic-simulation/economy/FactoryProducer.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <cstdint>
+
+#include "openvic-simulation/economy/Good.hpp"
+#include "openvic-simulation/economy/ProductionType.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+
+namespace OpenVic {
+	class FactoryProducer final {
+	private:
+		static constexpr uint8_t DAYS_OF_HISTORY = 7;
+		uint8_t PROPERTY(profit_history_current);
+		std::array<fixed_point_t, DAYS_OF_HISTORY> PROPERTY(daily_profit_history);
```

Comment:
`std::array<fixed_point_t, DAYS_OF_HISTORY>` is a good candidate for a type alias, for example:
```c++
using daily_profit_history_t = std::array<fixed_point_t, DAYS_OF_HISTORY>;
```